internal/abci/abci.go

```diff
@@ -428,6 +411,11 @@ func (a *AbciApp) Info(ctx context.Context, _ *abciTypes.RequestInfo) (*abciType
 		a.validatorAddressToPubKey[addr] = val.PubKey
 	}
 
+	err = a.txApp.ReloadDB(ctx)
```

Comment:
This certainly seems like the wrong place to call this. `Info` is called every time `/abci_info` is called. Not only do we not want to do a full DB reload here, but there is also a race condition with the engine's in-memory databases